ompl[app] build failure

**Host Environment**
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: ompl:x86-windows
  Vcpkg version: 2020.06.15-nohash

**To Reproduce**
Steps to reproduce the behavior:
`vcpkg install ompl[app]`

**Failure logs** 

> Computing installation plan...
> The following packages will be built and installed:
>     ompl[app,core]:x86-windows
> Detecting compiler hash for triplet x86-windows...
> Starting package 1/1: ompl:x86-windows
> Building package ompl[app,core]:x86-windows...
> Could not locate cached archive: C:\Users\gh7\AppData\Local\vcpkg\archives\eb\eb61858ccc2af961a34dad08c6886acf4dccd06d.zip
> -- Using cached E:/WS/vcpkg/downloads/omplapp-1.5.0.tar.gz
> -- Cleaning sources at E:/WS/vcpkg/buildtrees/ompl/src/1.5.0-05dfb62559.clean. Use --editable to skip cleaning for the packages you specify.
> -- Extracting source E:/WS/vcpkg/downloads/omplapp-1.5.0.tar.gz
> -- Using source at E:/WS/vcpkg/buildtrees/ompl/src/1.5.0-05dfb62559.clean
> -- Configuring x86-windows-dbg
> CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:85 (message):
>     Command failed: E:/WS/vcpkg/downloads/tools/cmake-3.18.4-windows/cmake-3.18.4-win32-x86/bin/cmake.exe E:/WS/vcpkg/buildtrees/ompl/src/1.5.0-05dfb62559.clean -DOMPL_VERSIONED_INSTALL=OFF -DOMPL_REGISTRATION=OFF -DOMPL_BUILD_DEMOS=OFF -DOMPL_BUILD_TESTS=OFF -DOMPL_BUILD_PYBINDINGS=OFF -DOMPL_BUILD_PYTESTS=OFF -DCMAKE_MAKE_PROGRAM=E:/WS/vcpkg/downloads/tools/ninja/1.10.1-windows/ninja.exe -DBUILD_SHARED_LIBS=ON -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=E:/WS/vcpkg/scripts/toolchains/windows.cmake -DVCPKG_TARGET_TRIPLET=x86-windows -DVCPKG_SET_CHARSET_FLAG=ON -DVCPKG_PLATFORM_TOOLSET=v141 -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE -DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_APPLOCAL_DEPS=OFF -DCMAKE_TOOLCHAIN_FILE=E:/WS/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON -DVCPKG_CXX_FLAGS= -DVCPKG_CXX_FLAGS_RELEASE= -DVCPKG_CXX_FLAGS_DEBUG= -DVCPKG_C_FLAGS= -DVCPKG_C_FLAGS_RELEASE= -DVCPKG_C_FLAGS_DEBUG= -DVCPKG_CRT_LINKAGE=dynamic -DVCPKG_LINKER_FLAGS= -DVCPKG_LINKER_FLAGS_RELEASE= -DVCPKG_LINKER_FLAGS_DEBUG= -DVCPKG_TARGET_ARCHITECTURE=x86 -DCMAKE_INSTALL_LIBDIR:STRING=lib -DCMAKE_INSTALL_BINDIR:STRING=bin -D_VCPKG_ROOT_DIR=E:/WS/vcpkg -D_VCPKG_INSTALLED_DIR=E:/WS/vcpkg/installed -DVCPKG_MANIFEST_INSTALL=OFF -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=E:/WS/vcpkg/packages/ompl_x86-windows/debug
>     Working Directory: E:/WS/vcpkg/buildtrees/ompl/x86-windows-dbg
>     Error code: 1
>     See logs for more information:
>       E:\WS\vcpkg\buildtrees\ompl\config-x86-windows-dbg-out.log
>       E:\WS\vcpkg\buildtrees\ompl\config-x86-windows-dbg-err.log
> 
> Call Stack (most recent call first):
>   scripts/cmake/vcpkg_configure_cmake.cmake:321 (vcpkg_execute_required_process)
>   ports/ompl/portfile.cmake:31 (vcpkg_configure_cmake)
>   scripts/ports.cmake:79 (include)
> 
> 
> Error: Building package ompl:x86-windows failed with: BUILD_FAILED

**Additional context**
`vcpkg install ompl` core lib only is OK but [app] generate the error
[config-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/5439604/config-x86-windows-dbg-out.log)
[config-x86-windows-dbg-err.log](https://github.com/microsoft/vcpkg/files/5439605/config-x86-windows-dbg-err.log)
[stdout-x86-windows.log](https://github.com/microsoft/vcpkg/files/5439609/stdout-x86-windows.log)

similar issue on CentOS Linux 7 (Core)

vcpkg install ompl[app] OK
vcpkg install ompl OK

> CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:85 (message):
>     Command failed: /home/moo/dev/vcpkg/downloads/tools/cmake-3.18.4-linux/cmake-3.18.4-Linux-x86_64/bin/cmake /home/moo/dev/vcpkg/buildtrees/ompl/src/1.5.0-05dfb62559.clean -DOMPL_VERSIONED_INSTALL=OFF -DOMPL_REGISTRATION=OFF -DOMPL_BUILD_DEMOS=OFF -DOMPL_BUILD_TESTS=OFF -DOMPL_BUILD_PYBINDINGS=OFF -DOMPL_BUILD_PYTESTS=OFF -DCMAKE_MAKE_PROGRAM=/home/moo/dev/vcpkg/downloads/tools/ninja-1.10.1-linux/ninja -DCMAKE_SYSTEM_NAME=Linux -DBUILD_SHARED_LIBS=OFF -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/home/moo/dev/vcpkg/scripts/toolchains/linux.cmake -DVCPKG_TARGET_TRIPLET=x64-linux -DVCPKG_SET_CHARSET_FLAG=ON -DVCPKG_PLATFORM_TOOLSET=external -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE -DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_APPLOCAL_DEPS=OFF -DCMAKE_TOOLCHAIN_FILE=/home/moo/dev/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON -DVCPKG_CXX_FLAGS= -DVCPKG_CXX_FLAGS_RELEASE= -DVCPKG_CXX_FLAGS_DEBUG= -DVCPKG_C_FLAGS= -DVCPKG_C_FLAGS_RELEASE= -DVCPKG_C_FLAGS_DEBUG= -DVCPKG_CRT_LINKAGE=dynamic -DVCPKG_LINKER_FLAGS= -DVCPKG_LINKER_FLAGS_RELEASE= -DVCPKG_LINKER_FLAGS_DEBUG= -DVCPKG_TARGET_ARCHITECTURE=x64 -DCMAKE_INSTALL_LIBDIR:STRING=lib -DCMAKE_INSTALL_BINDIR:STRING=bin -D_VCPKG_ROOT_DIR=/home/moo/dev/vcpkg -D_VCPKG_INSTALLED_DIR=/home/moo/dev/vcpkg/installed -DVCPKG_MANIFEST_INSTALL=OFF -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/home/moo/dev/vcpkg/packages/ompl_x64-linux/debug
>     Working Directory: /home/moo/dev/vcpkg/buildtrees/ompl/x64-linux-dbg
>     Error code: 1
>     See logs for more information:
>       /home/moo/dev/vcpkg/buildtrees/ompl/config-x64-linux-dbg-out.log
>       /home/moo/dev/vcpkg/buildtrees/ompl/config-x64-linux-dbg-err.log
> 
> Call Stack (most recent call first):
>   scripts/cmake/vcpkg_configure_cmake.cmake:321 (vcpkg_execute_required_process)
>   ports/ompl/portfile.cmake:31 (vcpkg_configure_cmake)
>   scripts/ports.cmake:79 (include)
> 
> 
> Error: Building package ompl:x64-linux failed with: BUILD_FAILED
> Please ensure you're using the latest portfiles with `./vcpkg update`, then
> submit an issue at https://github.com/Microsoft/vcpkg/issues including:
>   Package: ompl:x64-linux
>   Vcpkg version: 2020.06.15-unknownhash

@hortegilles, thanks for reporting this issue!

Failures:
CMake Error at CMakeLists.txt:177 (include_directories):
  include_directories given empty-string as include directory.

--- a/ports/ompl/CONTROL
+++ b/ports/ompl/CONTROL
@@ -1,5 +1,6 @@
 Source: ompl
 Version: 1.5.0
+Port-Version: 1
 Homepage: https://bitbucket.org/ompl
 Description: The Open Motion Planning Library, consists of many state-of-the-art sampling-based motion planning algorithms
 Build-Depends: boost-dynamic-bitset, boost-filesystem, boost-graph, boost-odeint, boost-program-options, boost-serialization, boost-system, boost-test, boost-ublas, boost-timer, eigen3

--- a/ports/ompl/fix_dependency.patch
+++ b/ports/ompl/fix_dependency.patch
@@ -1,0 +1,32 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 24be873..3b916d7 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -136,7 +136,7 @@ find_package(Drawstuff QUIET)
+ set_package_properties(assimp PROPERTIES
+     URL "http://assimp.org"
+     PURPOSE "Used in ompl_app for reading meshes representing robots and environments.")
+-find_package(assimp REQUIRED)
++find_package(assimp CONFIG REQUIRED)
+ set_package_properties(ccd PROPERTIES
+     URL "https://github.com/danfis/libccd"
+     PURPOSE "Collision detection library used by fcl.")
+@@ -178,8 +178,7 @@ include_directories(
+     "${OMPLAPP_INCLUDE_DIRS}"
+     "${OMPL_INCLUDE_DIRS}"
+     "${EIGEN3_INCLUDE_DIR}"
+-    "${FCL_INCLUDE_DIRS}"
+-    "${ASSIMP_INCLUDE_DIRS}")
++    "${FCL_INCLUDE_DIRS}")
+ 
+ # ROS installs fcl in /usr. In /usr/include/fcl/config.h it says octomap was
+ # enabled. Octomap is installed in /opt/ros/${ROS_DISTRO}/include (most
+@@ -199,7 +198,7 @@ set(OMPLAPP_LIBRARIES
+     ${Boost_FILESYSTEM_LIBRARY}
+     ${Boost_SYSTEM_LIBRARY}
+     ${OPENGL_LIBRARIES}
+-    ${ASSIMP_LIBRARIES}
++    assimp::assimp
+     ${FCL_LIBRARIES})
+ link_directories(${ASSIMP_LIBRARY_DIRS} ${CCD_LIBRARY_DIRS} ${OCTOMAP_LIBRARY_DIRS} ${FCL_LIBRARY_DIRS})
+ 

--- a/ports/ompl/portfile.cmake
+++ b/ports/ompl/portfile.cmake
@@ -2,12 +2,15 @@ vcpkg_buildpath_length_warning(37)
 
 set(OMPL_VERSION 1.5.0)
 
+set(FEATURE_PATCHES)
+
 if("app" IN_LIST FEATURES)
     vcpkg_download_distfile(ARCHIVE
         URLS "https://github.com/ompl/omplapp/releases/download/1.5.0/omplapp-1.5.0-Source.tar.gz"
         FILENAME "omplapp-${OMPL_VERSION}.tar.gz"
         SHA512 ad221b67146915cb63be6731ca2fa7d827d85b7fd175d87ee64c799311dfe4878935881b1ae6447357fdd178f70c9aa01b178e857261a8d8769affa1e58ed72b
     )
+    list(APPEND FEATURE_PATCHES fix_dependency.patch)
 else()
     vcpkg_download_distfile(ARCHIVE
         URLS "https://github.com/ompl/ompl/archive/1.5.0.tar.gz"
@@ -20,6 +23,7 @@ vcpkg_extract_source_archive_ex(
     OUT_SOURCE_PATH SOURCE_PATH
     ARCHIVE ${ARCHIVE}
     REF ${OMPL_VERSION}
+    PATCHES ${FEATURE_PATCHES}
 )
 
 # Based on selected features different files get downloaded, so use the following command instead of patch.

